src/main/java/chatbot/task/TaskManager.java

```diff
@@ -0,0 +1,216 @@
+    private TaskList tasks;
+    private Storage storage;
+
+
+    /**
+     * Constructor for this class.
+     */
+    public TaskManager() {
+        this.tasks = new TaskList();
+        this.storage = new Storage("./data/chatbot.txt");
+        this.tasks = storage.loadFromFile();
+    }
+
+    /**
+     * Add todo task into the list.
+     *
+     * @param t description of todo
+     * @throws ChatbotException when the description is empty
+     */
+    public void addTodo(String t) throws ChatbotException {
+        if (t == null || t.trim().isEmpty()) {
+            throw new ChatbotException("☹ OOPS!!! The description of a todo cannot be empty.");
+        }
+        Task task = new Todos(t);
+        tasks.add(task);
+        storage.saveToFile(tasks);
+        System.out.println("    ____________________________________________________________");
+        System.out.println("     Got it. I've added this task:");
+        System.out.println("     " + task.toString());
+        System.out.println("     Now you have " + tasks.size() + " tasks in the list.");
+        System.out.println("    ____________________________________________________________");
+    }
+
+    /**
+     * Add deadline to the list and print out.
+     *
+     * @param t description
+     * @param date date
+     * @throws ChatbotException when one of the param is empty
+     */
+    public void addDeadlines(String t, String date) throws ChatbotException {
+        if (t == null || t.trim().isEmpty()) {
+            throw new ChatbotException("☹ OOPS!!! The description of a deadlines cannot be empty.");
+        } else if (date == null || date.trim().isEmpty()) {
+            throw new ChatbotException("☹ OOPS!!! The date of a deadlines cannot be empty.");
+        }
+        try {
+            DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd/MM/yyyy HHmm");
+            LocalDateTime dateTime = LocalDateTime.parse(date, formatter);
+
+            Task task = new Deadlines(t, dateTime);
+            tasks.add(task);
+            storage.saveToFile(tasks);
+            System.out.println("    ____________________________________________________________");
+            System.out.println("     Got it. I've added this task:");
+            System.out.println("     " + task.toString());
+            System.out.println("     Now you have " + tasks.size() + " tasks in the list.");
+            System.out.println("    ____________________________________________________________");
+        } catch (DateTimeParseException e) {
+            System.out.println("    Invalid date/time format. Please enter in dd/MM/yyyy HHmm format.");
+        }
+    }
+
+    /**
+     * Add event to the list and print out.
+     *
+     * @param t description
+     * @param start starting date
+     * @param end end date
+     * @throws ChatbotException when one of the param is empty
+     */
+    public void addEvents(String t, String start, String end) throws ChatbotException {
```

Comment:
I like how you have accounted for all possible input errors here, and your other task parsing functions as well!